server/spectatord.cc

```diff
@@ -346,11 +346,15 @@ void Server::Start() {
     logger->info("statsd support is not enabled");
   }
 
-  prepare_socket_path(socket_path_);
-  LocalServer dgram_local{io_context, socket_path_, parser};
-  dgram_local.Start();
-  logger->info("Starting local server (dgram) on socket {}", socket_path_);
-  io_context.run();
+  if (socket_path_ != "**disabled**") {
+    prepare_socket_path(socket_path_);
+    LocalServer dgram_local{io_context, socket_path_, parser};
+    dgram_local.Start();
+    logger->info("Starting local server (dgram) on socket {}", socket_path_);
+    io_context.run();
```

Comment:
Not entirely sure if the io_context should be inside or outside the if block.

I think it should be included, because this is where the ASIO stuff is included, which we think may be a contributing factor to the XFS overlay issues.

Reply:
Correct. `io_context.run()` needs to be executed regardless of whether we're listening on the unix domain socket.

Reply:
Got it, will revise.